File: sdk-node/examples/subscriptions/uncomp-next-cycle.ts
/**
 * Remove a staged comp so the next renewal bills at full price again — the
 * inverse of `compNextCycle`. A no-op when no comp is pending, and allowed on
 * a subscription in any state.
 *
 * Run:
 *   npx tsx examples/subscriptions/uncomp-next-cycle.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const sub = await client.subscriptions.uncompNextCycle('sub_replace_with_real_id')

console.log(`subscription ${sub.id ?? '?'} [${sub.status ?? '?'}]`)
console.log(`  next renewal (${sub.currentPeriodEnd ?? '?'}) will bill at full price`)

File: sdk-python/examples/subscriptions/uncomp_next_cycle.py
"""Remove a staged comp so the next renewal bills at full price again.

The inverse of ``comp_next_cycle``. A no-op when no comp is pending, and allowed
on a subscription in any state.

Run with: python examples/subscriptions/uncomp_next_cycle.py
"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        sub = client.subscriptions.uncomp_next_cycle("sub_replace_with_real_id")
        print(f"subscription {sub.id} [{sub.status}]")
        print(f"  next renewal ({sub.current_period_end}) will bill at full price")


if __name__ == "__main__":
    main()
